shared/js/icons_helper.js

```diff
@@ -150,6 +149,39 @@
     });
   }
 
+  /**
+   * Same as above but set the image as the icon property of a gaia-app-icon
+   * element.
+   *
+   * @param icon {Object?}
+   * @param targetSize {number}
+   * @returns {Promise}
+   */
+  function setElementIcon(icon, targetSize) {
+    getIconBlob(icon.bookmark.url, targetSize, icon.bookmark, icon.bookmark)
+      .then(iconObj => {
+        if (iconObj.blob) {
+          icon.icon = iconObj.blob;
+        } else if (icon.bookmark.icon) {
+          return icon.bookmark.icon;
```

Comment:
Is there any point in returning here, rather than just embedding the below fetchIconBlob directly?